uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -336,8 +336,9 @@ private static function validateTOILPastDays($params, $absenceType) {
     $toDate = new DateTime($params['to_date']);
     $todayDate = new DateTime('today');
     $leaveDatesHasPastDates = $fromDate < $todayDate || $toDate < $todayDate;
+    $isACancellationRequest = in_array($params['status_id'], LeaveRequest::getCancelledStatuses());
 
-    if ($leaveDatesHasPastDates && !$absenceType->allow_accrue_in_the_past) {
+    if ($leaveDatesHasPastDates && !$absenceType->allow_accrue_in_the_past && !$isACancellationRequest) {
```

Comment:
I'm just wondering, would it be cleaner to have all validation at on layer (e.g. the service layer) instead of having some at service layer and some in BAO?

Reply:
Well, the issue is that the LeaveRequest service is mainly concerned with validations about who can do what. The finer grained validations usually reside on the LeaveRequest BAO. 
This help keep things consistent.

Reply:
I guess that makes sense, in that case it's not really validation in two places, in one place it's more like "security" and in the other it's "validation". Does that make sense?